lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -357,6 +357,16 @@ loadOpsToDistanceAndUse(scf::ForOp forOp) {
       continue;
     dfs(&op, 0, &op);
   }
+
+  // If the loop has numStages attribute, also consider pipelining other loads
+  // that are not directly used by dot ops.
+  if (forOp->hasAttr(tt::kNumStagesAttrName)) {
+    for (Operation &op : forOp.getBody()->without_terminator()) {
+      if (!isa<tt::LoadOp>(op))
```

Comment:
If op is `DotOp`, it's processed multiple times. First time Line 358 and second time Line 366. Not a big issue though

Reply:
Ops already processed will be bailed out in `dfs` due to the `seen.insert(op).second` check.

Reply:
Got it. Seems good then